spark/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/deltaMerge.scala

```diff
@@ -602,15 +602,26 @@ object DeltaMergeInto {
       // clause, then merge this schema with the target to give the final schema.
       def filterSchema(sourceSchema: StructType, basePath: Seq[String]): StructType =
         StructType(sourceSchema.flatMap { field =>
-          val fieldPath = basePath :+ field.name.toLowerCase(Locale.ROOT)
-          val childAssignedInMergeClause = assignments.exists(_.startsWith(fieldPath))
+          val fieldPath = basePath :+ field.name
+
+          // Helper method to check if a given field path is a prefix of another path. Delegates
+          // equality to conf.resolver to correctly handle case sensitivity.
+          def isPrefix(prefix: Seq[String], path: Seq[String]): Boolean =
```

Comment:
Do we want to add these methods to SchemaUtils?

Reply:
`deltaMerge` lives in package `catalyst` and `SchemaUtils` lives in `core`, we can't share the helpers between the two unfortunately.